src/aiokatcp/sensor.py

```diff
@@ -714,3 +715,69 @@ def copy(self) -> Dict[str, Sensor]:
     values.__doc__ = dict.values.__doc__
     items.__doc__ = dict.items.__doc__
     copy.__doc__ = dict.copy.__doc__
+
+
+# I'm ever so slightly nervous about inheritance and having a metaclass.
+class AggregateSensor(Sensor, metaclass=ABCMeta):
+    """A Sensor with its reading determined by several other Sensors.
+
+    This is an abstract class, the user must implement :meth:`_update_aggregate`
+    in order to define the logic of how the aggregate sensor's reading will be
+    determined by those of the sensor set to which it is targeted.
+
+    Parameters are all as per :class:`Sensor`, with the exception of
+    `target`, which is the :class:`SensorSet` from which the aggregated sensor
+    will determine its own reading.
+    """
+
+    def __init__(
+        self,
+        target: SensorSet,
+        sensor_type: Type[_T],
+        name: str,
+        description: str = "",
+        units: str = "",
+        default: _T = None,
+        initial_status: Sensor.Status = Sensor.Status.UNKNOWN,
+        *,
+        status_func: Callable[[_T], Sensor.Status] = _default_status_func,
+        auto_strategy: Optional["SensorSampler.Strategy"] = None,
+        auto_strategy_parameters: Iterable[Any] = (),
+    ) -> None:
+        Sensor.__init__(
+            self,
+            sensor_type=sensor_type,
+            name=name,
+            description=description,
+            units=units,
+            default=default,
+            initial_status=initial_status,
+            status_func=status_func,
+            auto_strategy=auto_strategy,
+            auto_strategy_parameters=auto_strategy_parameters,
+        )
+        self.target = target
+        for sensor in self.target.values():
+            if sensor is not self:  # This sensor might be in the set.
+                sensor.attach(self._update_aggregate)
+
+        self.target.add_add_callback(self._sensor_added)
+        self.target.add_remove_callback(self._sensor_removed)
```

Comment:
Actually that makes a lot of sense.
If the sensor-set is otherwise complete before this aggregate is created, and the underlying values don't change for a while, then it could be stuck on default for quite a long time, which may not be an accurate reflection of what's going on.